spring-core/src/main/java/org/springframework/core/observability/AutoTimer.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.core.observability;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import io.micrometer.core.annotation.Timed;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.Timer.Builder;
+
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Strategy that can be used to apply {@link Timer Timers} automatically instead of using
+ * {@link Timed @Timed}.
+ *
+ * @author Tadaya Tsuyukubo
+ * @author Stephane Nicoll
+ * @author Phillip Webb
+ * @since 6.0.0
+ */
+@FunctionalInterface
+public interface AutoTimer {
```

Comment:
Would this class be better in `io.micrometer`? Other than `CollectionUtils` there's no direct Spring connection.

Reply:
This is something we need to figure out. I think based on what we have right now, it would be but we need to check if we can/want to provide a different disable mechanism (for `Meter`s this mechanism is provided by `MeterFilter`s but it won't work in this use-case).